benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/SmallEntryCountLiteralMatcherBenchark.cs

```diff
@@ -20,78 +20,105 @@ public class SmallEntryCountLiteralMatcherBenchark : MatcherBenchmarkBase
+            _instruction = SetupMatcher(new InstructionMatcherBuilder());
+            _route = SetupMatcher(new RouteMatcherBuilder());
+            _tree = SetupMatcher(new TreeRouterMatcherBuilder());
+
+            _feature = new EndpointFeature();
+        }
+
+        private void SetupEndpoints()
+        {
+            _endpoints = new MatcherEndpoint[10];
+            _endpoints[0] = CreateEndpoint("/another-really-cool-entry");
+            _endpoints[1] = CreateEndpoint("/Some-Entry");
+            _endpoints[2] = CreateEndpoint("/a/path/with/more/segments");
+            _endpoints[3] = CreateEndpoint("/random/name");
+            _endpoints[4] = CreateEndpoint("/random/name2");
+            _endpoints[5] = CreateEndpoint("/random/name3");
+            _endpoints[6] = CreateEndpoint("/random/name4");
+            _endpoints[7] = CreateEndpoint("/plaintext1");
+            _endpoints[8] = CreateEndpoint("/plaintext2");
+            _endpoints[9] = CreateEndpoint("/plaintext");
+        }
+
+        private void SetupRequests()
+        {
             _requests = new HttpContext[1];
             _requests[0] = new DefaultHttpContext();
             _requests[0].RequestServices = CreateServices();
             _requests[0].Request.Path = "/plaintext";
-
-            _minimal = SetupMatcher(MinimalMatcher.CreateBuilder());
-            _dfa = SetupMatcher(DfaMatcher.CreateBuilder());
-            _instruction = SetupMatcher(InstructionMatcher.CreateBuilder());
-            _route = SetupMatcher(RouteMatcher.CreateBuilder());
-            _tree = SetupMatcher(TreeRouterMatcher.CreateBuilder());
-
-            _feature = new EndpointFeature();
         }
 
         // For this case we're specifically targeting the last entry to hit 'worst case'
         // performance for the matchers that scale linearly.
         private Matcher SetupMatcher(MatcherBuilder builder)
         {
-            builder.AddEntry("/another-really-cool-entry", null);
-            builder.AddEntry("/Some-Entry", null);
-            builder.AddEntry("/a/path/with/more/segments", null);
-            builder.AddEntry("/random/name", null);
-            builder.AddEntry("/random/name2", null);
-            builder.AddEntry("/random/name3", null);
-            builder.AddEntry("/random/name4", null);
-            builder.AddEntry("/plaintext1", null);
-            builder.AddEntry("/plaintext2", null);
-            builder.AddEntry("/plaintext", _endpoints[0]);
+            builder.AddEndpoint(_endpoints[0]);
+            builder.AddEndpoint(_endpoints[1]);
+            builder.AddEndpoint(_endpoints[2]);
+            builder.AddEndpoint(_endpoints[3]);
+            builder.AddEndpoint(_endpoints[4]);
+            builder.AddEndpoint(_endpoints[5]);
+            builder.AddEndpoint(_endpoints[6]);
+            builder.AddEndpoint(_endpoints[7]);
+            builder.AddEndpoint(_endpoints[8]);
+            builder.AddEndpoint(_endpoints[9]);
```

Comment:
I simplified the design of this a bit based on some of the other design choices we made elsewhere.